gateway/gateway.go

```diff
@@ -479,6 +470,29 @@ func (g *gateway) DoCommand(ctx context.Context, cmd map[string]interface{}) (ma
 	if _, ok := cmd["validate"]; ok {
 		return map[string]interface{}{"validate": 1}, nil
 	}
+	if deviceReq, ok := cmd["return_devices"]; ok {
+		resp := map[string]interface{}{}
+		deviceStr, _ := deviceReq.(string)
+		// look for a specific device if requested. If the device name is not a string we just return all devices.
+		if deviceStr != "" {
+			// check if the device has joined the network.
+			resp[deviceStr], ok = g.devices[deviceStr]
+			if ok {
+				resp["specific device currently joined"] = deviceStr
+			}
+			return nil, fmt.Errorf("device %v is not currently joined", deviceStr)
+		}
+		resp["all devices"] = deviceStr
```

Comment:
this is going to equal `{"all devices": ""}` right? What does that mean?

Reply:
was intended to signal to the user that we were in the all devices path. going to remove the individual sensor logic